File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "";
        int n = strs.size();
        if(strs[0].size()==0){
                return "";
            }
        for(int i=0; i<strs[0].size(); i++){
            char curr = strs[0][i];
            for(int j = 1; j<n; j++ ){
                if(strs[j][i] != curr || strs[j].size() < i)
                return ans;
            }
            ans+=curr;
        }
        return ans;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        if(strs.isEmpty() || strs[0].isEmpty()) return ""
        val n = strs.size
        val sb = StringBuilder()
        val first = strs[0]
        for(i in 0 until first.length){
            val curr = first[i]
            for( j in 0 until n){
                if( strs[j].length <= i || strs[j][i] != curr)
                    return sb.toString()
            }
            sb.append(curr)
        }
        return sb.toString()
    }
}